test/run_tests.pl

```diff
@@ -280,8 +290,8 @@ sub find_matching_tests {
 
 my $harness = $package->new(\%tapargs);
 my $ret =
-    $harness->runtests(map { [ abs2rel($_, rel2abs(curdir())), $tests{$_} ] }
-                       sort keys %tests);
+    $harness->runtests(map { [ abs2rel($_, rel2abs(curdir())), basename($tests{$_}) ] }
+                       sort { reorder($a) cmp reorder($b) } keys %tests);
```

Comment:
You *could* simplify further by changing things further:

-   all `%tests = map { $_ => $_ } @alltests;` to `%tests = map { $_ => 1 } @alltests;`
-   all `$tests{$test} = $test;` to `$tests{$test} = 1;`
-   these lines to:
    ``` perl
        $harness->runtests(map { [ abs2rel($_, rel2abs(curdir())), basename($_) ] }
                           sort { reorder($a) cmp reorder($b) } keys %tests);
    ```

The only reason to still have a `%tests` is that it guarantees uniqueness, and the `delete` line is much clearer than grepping stuff away from a list.

Reply:
Interesting simplification - done.